pkg/proposal/get.go

```diff
@@ -17,16 +17,115 @@ limitations under the License.
 package proposal
 
 import (
-	"github.com/bestchains/bc-cli/pkg/common"
+	"context"
+	"encoding/json"
+	"fmt"
+
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/cmd/get"
+
+	"github.com/bestchains/bc-cli/pkg/common"
+	"github.com/bestchains/bc-cli/pkg/utils"
 )
 
 func NewProposalGetCmd(option common.Options) *cobra.Command {
+	defaultPrintFlag := get.NewGetPrintFlags()
 	cmd := &cobra.Command{
-		Use: "",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return nil
+		Use: "proposal [NAME]",
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg, err := clientcmd.BuildConfigFromKubeconfigGetter("", common.InKubeGetter)
```

Comment:
Similar logic appears multiple times, regardless of the resources being obtained. 
Can we abstract the similar parts into a function?